src/com/amazon/ion/IonSequence.java

```diff
@@ -410,6 +410,73 @@ public void add(int index, IonValue child)
      * was created.
      * </p>
      *
+     * The implementation of {@link List<IonValue>} returned by this method implements {@link Object#equals(Object)}
+     * and {@link Object#hashCode()}, with some caveats to be aware of illustrated below.
```

Comment:
... and {@link Object#hashCode()} per the specification of these methods. However,  the existing implementation of `IonSequence`  does **not** provide a specification compliant `equals()` and `hashCode()` which results to the following caveats